pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -105,10 +105,14 @@ public void runJob() {
             processSegments();
           } catch (EarlyTerminationException e) {
             // Early-terminated by interruption (canceled by the main thread)
-          } catch (Exception e) {
-            // Caught exception, skip processing the remaining segments
-            LOGGER.error("Caught exception while processing query: " + _queryContext, e);
-            onException(e);
+          } catch (Throwable t) {
+            // Caught exception/error, skip processing the remaining segments
+            if (t instanceof Exception) {
+              LOGGER.error("Caught exception while processing query: " + _queryContext, t);
+            } else {
+              LOGGER.error("Caught serious error while processing query: " + _queryContext, t);
```

Comment:
Not sure if I follow the distinction here between an exception and a serious error

Reply:
Usually Java doesn't want user code to catch the `Error` which indicates some serious problems (see `Error.java` for more details). But for our specific case, we need to catch them in the sub-thread, or the thread will die without inserting the exceptions, and can cause unexpected behavior. Will add some comments